Error when using JDK11

Hi, thank you for this awesome library!

I'm having some trouble after upgrading to java 11 and clojure 1.10.0.rc-4. I can't run `lein test` or `lein eftest`, and instead I get the following stacktrace:

```
16:06:42.723 INFO  medino-web.handler - Selmer template cache is OFF.
Exception in thread "main" java.lang.ExceptionInInitializerError
	at clojure.main.<clinit>(main.java:20)
Caused by: Syntax error compiling deftype* at (clojure/core/rrb_vector/rrbt.clj:282:1).
	at clojure.lang.Compiler.analyzeSeq(Compiler.java:7114)
	....
	at clojure.lang.RT.loadResourceScript(RT.java:368)
	at clojure.lang.RT.maybeLoadResourceScript(RT.java:364)
	at clojure.lang.RT.doInit(RT.java:485)
	at clojure.lang.RT.<clinit>(RT.java:338)
	... 1 more
Caused by: java.lang.IllegalArgumentException: Must hint overloaded method: toArray
	at clojure.lang.Compiler$NewInstanceMethod.parse(Compiler.java:8496)
	at clojure.lang.Compiler$NewInstanceExpr.build(Compiler.java:8058)
	at clojure.lang.Compiler$NewInstanceExpr$DeftypeParser.parse(Compiler.java:7934)
	at clojure.lang.Compiler.analyzeSeq(Compiler.java:7106)
	... 244 more
Tests failed.
```

I believe this is caused by this bug in `core.rrb-vector.`: https://dev.clojure.org/jira/browse/CLJ-2374
which is a dependency of puget.

Putting `[mvxcvi/puget "1.0.3"]` in my dependency list resolves the issue, so I belive upgrading that in the project.clj files should resolve the issue.

Happy to make a pull request if that helps at all.
Thanks!

--- a/eftest/project.clj
+++ b/eftest/project.clj
@@ -7,4 +7,4 @@
                  [org.clojure/tools.namespace "0.2.11"]
                  [progrock "0.1.2"]
                  [io.aviso/pretty "0.1.34"]
-                 [mvxcvi/puget "1.0.2"]])
+                 [mvxcvi/puget "1.0.3"]])

